bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableOptionsFactory.java

```diff
@@ -550,11 +599,451 @@ private static RetryOptions createRetryOptions(Configuration configuration) {
+    try {
+      PrivateKey privateKey =
+          SecurityUtils.loadPrivateKeyFromKeyStore(
+              SecurityUtils.getPkcs12KeyStore(),
+              new FileInputStream(privateKeyFile),
+              "notasecret",
+              "privatekey",
+              "notasecret");
+
+      return ServiceAccountJwtAccessCredentials.newBuilder()
+          .setClientEmail(serviceAccountEmail)
+          .setPrivateKey(privateKey)
+          .build();
+    } catch (GeneralSecurityException exception) {
+      throw new RuntimeException("exception while retrieving credentials", exception);
+    }
+  }
+
+  /** Creates {@link TransportChannelProvider} for plaintext negotiation type. */
+  private static TransportChannelProvider buildPlainTextChannelProvider(
+      String endpoint, int channelCount) {
+    return defaultGrpcTransportProviderBuilder()
+        .setEndpoint(endpoint)
+        .setPoolSize(channelCount)
+        .setChannelConfigurator(
+            new ApiFunction<ManagedChannelBuilder, ManagedChannelBuilder>() {
+              @Override
+              public ManagedChannelBuilder apply(ManagedChannelBuilder channelBuilder) {
+                return channelBuilder.usePlaintext();
+              }
+            })
+        .build();
+  }
+
+  /** Creates {@link Set} of {@link StatusCode.Code} from {@link Status.Code} */
+  private static Set<StatusCode.Code> buildRetryCodes(Configuration configuration) {
+    ImmutableSet.Builder<StatusCode.Code> statusCodeBuilder = ImmutableSet.builder();
+
+    for (Status.Code retryCode : DEFAULT_ENABLE_GRPC_RETRIES_SET) {
+      statusCodeBuilder.add(GrpcStatusCode.of(retryCode).getCode());
+    }
+
+    String retryCodes = configuration.get(ADDITIONAL_RETRY_CODES, "");
+    String[] codes = retryCodes.split(",");
+    for (String stringCode : codes) {
+      String trimmed = stringCode.trim();
+      if (trimmed.isEmpty()) {
+        continue;
+      }
+      StatusCode.Code code = StatusCode.Code.valueOf(trimmed);
+      Preconditions.checkNotNull(code, "Code " + stringCode + " not found.");
+      statusCodeBuilder.add(code);
+      LOG.debug("gRPC retry on: %s", stringCode);
+    }
+
+    return statusCodeBuilder.build();
+  }
+
+  /** Creates {@link RetrySettings} for non-streaming idempotent method. */
+  private static RetrySettings buildIdempotentRetrySettings(
+      RetrySettings originalRetrySettings, Configuration configuration) {
+
+    RetrySettings.Builder retryBuilder = originalRetrySettings.toBuilder();
+    if (configuration.getBoolean(ALLOW_NO_TIMESTAMP_RETRIES_KEY, false)) {
+      throw new UnsupportedOperationException("Retries without Timestamp is not supported yet.");
+    }
+
+    boolean useTimeOut = configuration.getBoolean(BIGTABLE_USE_TIMEOUTS_KEY, USE_TIMEOUT_DEFAULT);
+    int shortRpcTimeoutMs =
+        configuration.getInt(BIGTABLE_RPC_TIMEOUT_MS_KEY, SHORT_TIMEOUT_MS_DEFAULT);
+    Duration rpcTimeout = ofMillis(useTimeOut ? shortRpcTimeoutMs : RPC_DEADLINE_MS);
```

Comment:
This looks wrong. If I understand correctly, useTimeout = false means that deadlines are supposed to be disabled, but here you are setting it to 6 minutes?

Reply:
Oh, I followed the existing conversion from BigtableOptions to BigtableDataSettings.
https://github.com/googleapis/java-bigtable-hbase/blob/41a3d98cb4561917ac91e1ea9207b99e1f3e0067/bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/BigtableVeneerSettingsFactory.java#L275-L277
If this seems wrong then I can quickly open a PR to fix that as well